### PR TITLE
[2.19.x] Upgrading to Solr 8.8.1

### PR DESCRIPTION
**Pull request by kcwire**
#### What does this PR do?
Upgrades DDF to use Solr 8.8.1.
#### Who is reviewing it? 
@millerw8 
@pklinef 
@andrewzimmer 
@Lambeaux 

#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewzimmer
@cassandrabailey293

#### How should this be tested?
Build and run DDF as usual. Ingest, update, delete sample data.

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrClientAdapter.java [hunk 1]
```diff
@@ -443,10 +443,20 @@ public void whenAvailable(Initializer initializer) {
...
+  // which goes through getProxiedClient() which would throw back an unavailable error instead of
   // returning the response
   public SolrPingResponse ping() throws SolrServerException, IOException {
-    return ping("from the API");
+    LOGGER.debug("Solr: pinging the client from the API");
+    return clientPing(null);
+  }
+
+  @Override
+  // overriden to always send the ping to the client; avoiding the intercept in handle()
```
Comment (bellcc) [inline platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrClientAdapter.java:454]: ✏️ You've got a typo right here. There are two d's in `overridden`

File: pom.xml [hunk 2]
```diff
@@ -327,6 +328,9 @@
         <tink.version>1.2.2</tink.version>
         <org.json.version>20170516</org.json.version>
         <ehcache.version>2.10.6</ehcache.version>
+
+        <!-- temporaru solr download url -->
```
Comment (rymach) [inline pom.xml:332]:
```suggestion
        <!-- temporary solr download url -->
```

**Review by rymach — APPROVED** [review]

**Comment by rymach** [comment]
build now

File: catalog/core/catalog-core-solr/pom.xml [hunk 3]
```diff
@@ -29,6 +29,11 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-simple</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (mojogitoverhere) [inline catalog/core/catalog-core-solr/pom.xml:36]: Just curious, why was this moved up?

File: platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java [hunk 4]
```diff
@@ -766,6 +767,16 @@ public UpdateResponse deleteByQuery(String collection, String query, int commitW
   public UpdateResponse deleteByQuery(String query, int commitWithinMs)
       throws SolrServerException, IOException;
 
+  /**
+   * Issues a ping request to check if the collection's replicas are alive
+   *
+   * @param collection the Solr collection to ping
+   * @return a {@link SolrPingResponse} containing the response from the server
+   * @throws IOException If there is a low-level I/O error
```
Comment (mojogitoverhere) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java:775]: consistency is king 👑 
```suggestion
   * @throws IOException if there is a low-level I/O error
```

File: platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java [hunk 5]
```diff
@@ -847,6 +858,19 @@ public QueryResponse queryAndStreamResponse(
       String collection, SolrParams params, StreamingResponseCallback callback)
       throws SolrServerException, IOException;
 
+  /**
+   * @param collection the Solr collection to query
+   * @param params an object holding all key/value parameters to send along the request
+   * @param callback the callback to stream results to
+   * @return a {@link QueryResponse} containing the response from the server
+   * @throws IOException If there is a low-level I/O error
```
Comment (mojogitoverhere) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java:866]:
```suggestion
   * @throws IOException if there is a low-level I/O error
```

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java [hunk 6]
```diff
@@ -86,19 +86,22 @@ SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
...
         createCollection(collection, client);
       } catch (SolrFactoryException e) {
-        LOGGER.debug("Solr({}): Unable to create collection on Solr Cloud", collection, e);
+        LOGGER.debug("Solr({}): Unable to create collection on SolrCloud", collection, e);
         return null;
       }
 
       client.setDefaultCollection(collection);
       return closer.returning(client);
+    } catch (LinkageError | Exception e) {
```
Comment (mojogitoverhere) [inline platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java:102]: Is there something more specific we can catch besides `Exception`?

**Review by mojogitoverhere — APPROVED** [review]

**Review by andrewzimmer — APPROVED** [review]

File: catalog/core/catalog-core-solr/pom.xml (hunk 3 above)
Comment (pklinef) [reply catalog/core/catalog-core-solr/pom.xml:36]: Usually this is done to have a dependency take priority over another dependency's transitive dependencies.

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java [hunk 7]
```diff
@@ -79,7 +79,7 @@
 
   private static final String REQUEST_MUST_NOT_BE_NULL_MESSAGE = "Request must not be null";
 
-  static final int MAX_BOOLEAN_CLAUSES = 2048;
+  static final int MAX_BOOLEAN_CLAUSES = 1024;
```
[the remark this answers is not in the document]
Comment (pklinef) [reply catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java:82]: We might consider making this a configuration property like Solr does.

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java (hunk 6 above)
Comment (pklinef) [reply platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java:102]: I think we had something more specific before but any unhandled exceptions from the factory will cause failsafe to exit and stop retrying. So that is why this is as aggressive as it is on catching exceptions.

**Review by pklinef — APPROVED** [review]

**Review by pklinef — COMMENTED** [review]
Solr 8.8 upgraded commons-io from 2.6 to 2.8. We should update `solr-dependencies`.

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (kcwire) [reply catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java:82]: I think it's best to propose this as an improvement rather than lump it into this ticket.

**Comment by kcwire** [comment]
Addressed all comments and rebased to latest.

**Comment by kcwire** [comment]
build now
